features/communi_functions.prf

```diff
@@ -9,7 +9,7 @@ defineTest(communiAddLib) {
     isEmpty(LIBNAME):error(LIBNAME must be set)
     LIBS *= -L$$BUILD_TREE/lib
     LIBS *= -l$$LIBNAME
-    PRE_TARGETDEPS *= $$BUILD_TREE/lib/$${QMAKE_PREFIX_STATICLIB}$${LIBNAME}.$${QMAKE_EXTENSION_STATICLIB}
+    PRE_TARGETDEPS *= $$BUILD_TREE/lib/lib$${LIBNAME}.$${QMAKE_EXTENSION_SHLIB}
```

Comment:
looks like QMAKE_EXTENSION_SHLIB can be empty, so you could replace the QMAKE_EXTENSION_STATICLIB hack at the beginning of the file with

    isEmpty(QMAKE_EXTENSION_SHLIB):QMAKE_EXTENSION_SHLIB = so